Enable the plugin to assume a role

sihil opened the pull request:
We have a requirement to get data from a Kinesis stream in a different AWS account.

This PR adds a new `role_arn` config option that takes an AWS role ARN that will be assumed. This role is assumed from either the default or profile credentials. The newly created credential provider is used only for accessing the kinesis stream. Both dynamo and cloudwatch access by the KCL continue to be done using the underlying credential provider and so these resources are in the home account rather than the account containing the stream.

sihil commented:
I'll figure out the CLA.

Based on the other PRs I don't think this is responsible for the broken build. Is that something that is likely to be looked at?

sihil commented:
I've now rebased this against master as the build has been fixed.

Also the creation of #47 has made me notice that #34 exists. Don't mind which gets merged, but this has been running in PROD for several months now.

sihil commented:
Hi @robbavey - I've noticed a number of PRs subsequent to this get reviewed and merged. Do I need to do something to get this picked up by someone?

robbavey commented:
@sihil Thanks for the reminder - I missed this when I was merging the other PRs. Let me take a look and get back to you

robbavey commented on lib/logstash/inputs/kinesis.rb:
```diff
@@ -78,6 +81,15 @@ def register
+
+    # If a role ARN is set then assume the role as a new layer over the credentials already created
+    unless @role_arn.nil?
+      session_id = "worker" + worker_id
```
Is it possible to add an extra configuration parameter `role_session_name` for the sake of consistency with other logstash aws plugins ([this is in the common aws mixin gem shared by most aws plugins](https://github.com/logstash-plugins/logstash-mixin-aws/blob/f23c0bf56db5cc4bf4a4b07c6d618b94a5c83e94/lib/logstash/plugin_mixins/aws_config/generic.rb#L38))

sihil replied:
Useful to know. Will look at this today and see what the best approach is.

sihil commented:
@robbavey I've addressed your comment. I think the build is failing on master too but I haven't figured out why. I'm also going to deploy my changes to our production environment to confirm they work as expected but can't see why they wouldn't. (Although the more recent KCL requires `kinesis:ListShards` permission)

sihil commented:
Thanks @robbavey! This has been on our prod stack for a few hours now without any issues.

robbavey commented:
@sihil I've added your commits to a new PR (#57), which includes the doc changes, version bumps, etc that we need in order to be able to publish the gem.

robbavey commented:
@sihil PR #57 which incorporated these changes has been merged, and the plugin published. Thank you again for your contribution!
